test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/BufferedWriterTest.scala

```diff
@@ -0,0 +1,44 @@
+package org.scalajs.testsuite.javalib.io
+
+import java.io._
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.testsuite.utils.AssertThrows.assertThrows
+
+class BufferedWriterTest {
```

Comment:
We'll probably have to add more tests, notably (probably not a complete list, sorry):
- operations on closed writers throw
- index out of bounds use cases
- corner cases pointed out in the implementation review (notably flushing behavior)
- writing to a buffered writer multiple times (with different buffer boundary conditions).

Reply:
I have added more test cases.